cmd/skopeo/delete.go

```diff
@@ -3,44 +3,52 @@ package main
 import (
 	"errors"
 	"fmt"
-	"io"
 	"strings"
 
 	"github.com/containers/image/v5/transports"
 	"github.com/containers/image/v5/transports/alltransports"
-	"github.com/urfave/cli"
+	"github.com/spf13/cobra"
 )
 
 type deleteOptions struct {
 	global *globalOptions
 	image  *imageOptions
 }
 
-func deleteCmd(global *globalOptions) cli.Command {
-	sharedFlags, sharedOpts := sharedImageFlags()
-	imageFlags, imageOpts := imageFlags(global, sharedOpts, "", "")
-	opts := deleteOptions{
-		global: global,
-		image:  imageOpts,
-	}
-	return cli.Command{
-		Name:  "delete",
-		Usage: "Delete image IMAGE-NAME",
-		Description: fmt.Sprintf(`
-	Delete an "IMAGE_NAME" from a transport
+func deleteCommand(globalFlagResults *globalOptions) *cobra.Command {
+	var (
+		opts              deleteOptions
```

Comment:
Why was this changed from just using a struct constructor? Now the reader has to worry about uses of the uninitialized value.

Reply:
it will be initialized in the flag definition, other fields remain zero value.

Reply:
Sure, it is functionally equivalent; my point is that the period of time where the pointers are uninitialized is introduced unnecessarily — extra work to make the code less readable.